File: src/main/java/com/vc/easy/L21.scala
package com.vc.easy

class ListNode21(var _x: Int = 0) {
  var next: ListNode21 = null
  var x: Int = _x
}


object L21 {
  def mergeTwoLists(l1Param: ListNode21, l2Param: ListNode21): ListNode21 = {
    var l1 = l1Param
    var l2 = l2Param
    val res = new ListNode21(-1)
    var prev = res
    while(l1 != null && l2 != null) {
      if(l1.x < l2.x) {
        prev.next = l1
        l1 = l1.next
        prev = prev.next
      }
      else {
        prev.next = l2
        l2 = l2.next
        prev = prev.next
      }
    }

    while(l1 != null) {
      prev.next = l1
      l1 = l1.next
      prev = prev.next
    }

    while(l2 != null) {
      prev.next = l2
      l2 = l2.next
      prev = prev.next
    }

    res.next
  }
}

File: src/main/java/com/vc/easy/L21.java
package com.vc.easy;

import com.vc.hard.ListNode;

class L21 {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2) {
        final ListNode list = new ListNode(-1);
        ListNode res = list;

        while(l1 != null && l2 != null) {
            if(l1.val < l2.val) {
                res.next = l1;
                l1 = l1.next;
                res = res.next;
            }
            else {
                res.next = l2;
                l2 = l2.next;
                res = res.next;
            }
        }

        while(l1 != null) {
            res.next = l1;
            l1 = l1.next;
            res = res.next;
        }

        while(l2 != null) {
            res.next = l2;
            l2 = l2.next;
            res = res.next;
        }

        return list.next;
    }
}
